CMakeLists.txt

```diff
@@ -6,13 +6,18 @@ find_package(Threads REQUIRED)
 
 set(CMAKE_FIND_PACKAGE_RESOLVE_SYMLINKS TRUE)
 set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 if(NOT APPLE)
     set(CMAKE_CXX_VISIBILITY_PRESET hidden)
     set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
 endif()
 
+option(ENABLE_WERROR "Treat all warnings as errors" FALSE)
+if(NOT MSVC AND ENABLE_WERROR)
```

Comment:
Might it make more sense to leave this option available on windows, and just explicitly disable it in the windows CI Job? we might want to manually test enabling this and going through all the errors at some point.

Reply:
Yeah, that makes way more sense.